agent-discoveries-backend/src/routes/agent.ts

```diff
@@ -0,0 +1,101 @@
+import { Router } from 'express'
+import { Knex } from 'knex'
+import createAuthMiddleware from '../middleware/auth'
+
+function createRouter(db: Knex) {
+  const auth = createAuthMiddleware(db)
+  const router = Router()
+
+  router.post('/', auth.isAdmin, async (req, res) => {
+    // this unpacking and re-packing is to make sure that we only insert
+    // the fields we want and not any that the user may sneak into the
+    // request body
+    const { callSign, firstName, lastName, dateOfBirth, agentRank } =
+      req.body
+
+    const newAgent = {
+      callSign,
+      firstName,
+      lastName,
+      dateOfBirth,
+      agentRank,
+    }
+
+    await db('agents').insert([newAgent])
+
+    res.status(200).send()
+  })
+
+  router.get('/:callSign', auth.isLoggedIn, async (req, res) => {
+    var query = db('agents')
+      .select('agentId', 'callSign', 'agentRank')
+      .where({ callSign: req.params.callSign })
+
+    try {
+      const agents = await query
+      res.status(200).json(agents)
+    } catch (e) {
+      res.status(400).send()
+    }
+  })
+
+  router.put('/update', auth.isLoggedIn, async (req, res) => {
+    const { callSign } = req.body
+
+    if (req.session.user?.agentId === undefined) {
+      return res.status(400).send()
+    }
+
+    try {
+      await db('agents')
+        .where({ agentId: req.session.user?.agentId })
+        .update({ callSign })
+      res.status(200).send()
+    } catch (e) {
+      res.status(400).send()
+    }
+  })
+
+  router.put('/delete', auth.isLoggedIn, async (req, res) => {
```

Comment:
Ooh actually - we could do 
```
router.delete('/:callsign', ...
```
or
```
router.delete('/:agentId', ...
```
which would be a bit more of a RESTful style than having a /delete endpoint :)

Reply:
Certainly can do. I mostly copied this from the users CRUD PR so I can go back and update that one afterwards

Reply:
Done